tests/test_fuzzy_checker.py

```diff
@@ -1,7 +1,247 @@
+    numerator: int, denominator: int, target_proportion: float, match: str
+) -> None:
+    with pytest.raises(AssertionError, match=match):
+        FuzzyChecker().fuzzy_assert_proportion(numerator, denominator, target_proportion)
+
+
+def test_small_sample_size_fuzzy_assert_proportion(caplog: LogCaptureFixture) -> None:
+    FuzzyChecker().fuzzy_assert_proportion(1, 10, 0.1)
+    assert "Sample size too small" in caplog.text
+
+
+def test_not_conclusive_fuzzy_assert_proportion(caplog: LogCaptureFixture) -> None:
+    # This test verifies we will pass, then be inconclusive, then fail.
+    # The numbers used in this test are arbitrary but are intended to be conservative
+    # estimates of the number of iterations needed to reach each state
+    # Creating an instance here allows us to cache some of the computation for the while loop
+    fuzzy_checker = FuzzyChecker()
+    i = 1_000
+    while True:
+        caplog.clear()
+        fuzzy_checker.fuzzy_assert_proportion(i, 10_000, 0.1)
+        if "is not conclusive" in caplog.text:
+            assert i > 1050
+            break
+        if i > 1_200:
+            raise RuntimeError("Test did not reach the expected warning")
+        i += 1
+
+    while True:
+        caplog.clear()
+        try:
+            fuzzy_checker.fuzzy_assert_proportion(i, 10_000, 0.1)
+            assert "is not conclusive" in caplog.text
+        except AssertionError as e:
+            assert "is significantly greater" in str(e)
+            assert i > 1_100
+            break
+        if i > 1_300:
+            raise RuntimeError("Test did not reach the expected warning")
+        i += 1
+
+
+@pytest.mark.parametrize(
+    "numerator",
+    [
+        1_008,
+        10_008,
+        100_008,
+        1_000_008,
+    ],
+)
+@pytest.mark.parametrize("denominator", OBSERVED_DENOMINATORS)
+def test__calculate_bayes_factor(numerator: int, denominator: int) -> None:
+    if numerator > denominator:
+        pytest.skip("Numerator cannot be greater than denominator.")
+    # Parametrize rv_discrete for no bug distribution
+    # I am keeping the defaults for the bug distribution to remain 0.5 for alpha and beta
+    bug_issue_distribution = scipy.stats.betabinom(a=0.5, b=0.5, n=denominator)
+    no_bug_issue_distribution = scipy.stats.binom(p=TARGET_PROPORTION, n=denominator)
+    bayes_factor = FuzzyChecker()._calculate_bayes_factor(
+        numerator, bug_issue_distribution, no_bug_issue_distribution
+    )
+    assert isinstance(bayes_factor, float)
+    assert bayes_factor > 0
+
+
+def test_zero_division__calculate_bayes_factor(caplog: LogCaptureFixture) -> None:
+    # This is just testing that we will hit a zero division error or floating point error
+    # and handle it correctly.
+    # We want the case where we observe a proportion that indicates an event is very likely
+    # but we expect it to be very unlikely.
+    numerator = 10_000_000
+    denominator = 10_000_000 - 1
```

Comment:
Isn't this impossible, since the numerator is larger than the denominator? I'm actually surprised this isn't throwing an error!

Reply:
Yea that was a big typo. But in the calculation of the bayes function there are only distributions and a numerator. So the numerator and denominator are unknown to each other. In fuzzy assert proportion we check that the numerator is less than the denominator so that is why it didn't throw an error.

Reply:
To be clear: I didn't expect the error to come from our code. I guess Scipy considers this not to be an error, but just something with 0 probability.